odoo/addons/base/models/ir_qweb.py

```diff
@@ -517,6 +517,10 @@ def keep_query(*keep_params, **additional_params):
 ###        QWebException         ###
 ####################################
 
+
+QWEB_ERROR_CLASSES = {}
```

Comment:
Does this simplify the code that much?
We used to have `__cause__` to unwrap the real error. Now that you are yielding, you can just wrap the error once at the end and the external calls can unwrap it. You could even add a property to ease `__cause__ or __context__`.

It feels a little bit strange to have `e = QWebValidationError` and `x = ValidationError` with `e.__cause__ == x`, you could think that `e` is the original exception, but some attributes (other than message) could not be properly initialized. This is not a real issue here because the errors chosen have just a message.